### PR TITLE
add automatically translated string literals

### PR DESCRIPTION
**Pull request by john-tornblom**
I've done my best at determining which strings should be translated, but keep an eye our for errors.

### CODE REVIEW

File: Source/DiabloUI/selhero.cpp [hunk 1]
```diff
@@ -403,81 +403,81 @@ const char *SelheroGenerateName(HeroClass heroClass)
...
-		    "Qarak",
-		    "Born",
-		    "Cathan",
-		    "Halbu",
-		    "Lenalas",
-		    "Maximus",
-		    "Vane",
-		    "Myrdgar",
-		    "Rothat",
+		    _("Aidan"),
```
Comment (AJenbo) [inline Source/DiabloUI/selhero.cpp:406]: Names should not be translated :)

File: Source/control.cpp [hunk 2]
```diff
@@ -1898,23 +1899,23 @@ void CheckSBook()
...
-	sprintf(tempstr, "You have %u gold", initialDropGoldValue);
+	sprintf(tempstr, _("You have %u gold"), initialDropGoldValue);
 	ADD_PlrStringXY(out, 366, 87, 600, tempstr, COL_GOLD);
-	sprintf(tempstr, "%s.  How many do", get_pieces_str(initialDropGoldValue));
+	sprintf(tempstr, _("%s.  How many do"), get_pieces_str(initialDropGoldValue));
 	ADD_PlrStringXY(out, 366, 103, 600, tempstr, COL_GOLD);
 	ADD_PlrStringXY(out, 366, 121, 600, "you want to remove?", COL_GOLD);
 	if (amount > 0) {
-		sprintf(tempstr, "%u", amount);
+		sprintf(tempstr, _("%u"), amount);
```
Comment (AJenbo) [inline Source/control.cpp:1918]:
```suggestion
		sprintf(tempstr, "%u", amount);
```

File: Source/error.cpp [hunk 3]
```diff
@@ -17,61 +18,61 @@ uint8_t msgcnt;
...
-	"Give and you shall receive.",
-	"Some experience is gained by touch.",
-	"There's no place like home.",
-	"Spiritual energy is restored.",
-	"You feel more agile.",
-	"You feel stronger.",
-	"You feel wiser.",
-	"You feel refreshed.",
-	"That which can break will.",
+	N_(""),
```
Comment (AJenbo) [inline Source/error.cpp:21]:
```suggestion
	"",
```

File: Source/items.cpp [hunk 4]
```diff
@@ -3452,292 +3453,292 @@ void PrintItemPower(char plidx, ItemStruct *x)
...
+			sprintf(tempstr, _("lightning damage: %i-%i"), x->_iFMinDam, x->_iFMaxDam);
 		break;
 	case IPL_ADDMANAAC:
-		strcpy(tempstr, "charged bolts on hits");
+		strcpy(tempstr, _("charged bolts on hits"));
 		break;
 	case IPL_FIRERESCLVL:
 		if (x->_iPLFR <= 0)
-			sprintf(tempstr, " ");
+			sprintf(tempstr, _(" "));
```
Comment (AJenbo) [inline Source/items.cpp:3706]:
```suggestion
			sprintf(tempstr, " ");
```

File: Source/stores.cpp [hunk 5]
```diff
@@ -1182,7 +1182,7 @@ void S_StartBarMaid()
 {
 	stextsize = false;
 	stextscrl = false;
-	AddSText(0, 2, true, "Gillian", COL_GOLD, false);
+	AddSText(0, 2, true, _("Gillian"), COL_GOLD, false);
```
Comment (AJenbo) [inline Source/stores.cpp:1185]: Gillian appears without title, so nothing to translate

```suggestion
	AddSText(0, 2, true, "Gillian", COL_GOLD, false);
```

File: Source/textdat.cpp [hunk 6]
```diff
@@ -4,495 +4,496 @@
...
 	    true, TSFX_STORY38 },
-	{ "Lazarus was the Archbishop who led many of the townspeople into the labyrinth. I lost many good friends that day, and Lazarus never returned. I suppose he was killed along with most of the others. If you would do me a favor, good master - please do not talk to Farnham about that day. |",
+	{ N_("Lazarus was the Archbishop who led many of the townspeople into the labyrinth. I lost many good friends that day, and Lazarus never returned. I suppose he was killed along with most of the others. If you would do me a favor, good master - please do not talk to Farnham about that day. |"),
 	    true, TSFX_TAVERN1 },
-	{ "|", true, TSFX_STORY38 },
-	{ "|", true, TSFX_STORY38 },
-	{ "I was shocked when I heard of what the townspeople were planning to do that night. I thought that of all people, Lazarus would have had more sense than that. He was an Archbishop, and always seemed to care so much for the townsfolk of Tristram. So many were injured, I could not save them all... |",
+	{ N_("|"), true, TSFX_STORY38 },
+	{ N_("|"), true, TSFX_STORY38 },
```
Comment (AJenbo) [inline Source/textdat.cpp:70]: Moo
```suggestion
	{ "|", true, TSFX_STORY38 },
	{ "|", true, TSFX_STORY38 },
```

File: Source/textdat.cpp [hunk 7]
```diff
@@ -4,495 +4,496 @@
...
 	    true, TSFX_WITCH3 },
-	{ "Yes, the righteous Lazarus, who was sooo effective against those monsters down there. Didn't help save my leg, did it? Look, I'll give you a free piece of advice. Ask Farnham, he was there. |",
+	{ N_("Yes, the righteous Lazarus, who was sooo effective against those monsters down there. Didn't help save my leg, did it? Look, I'll give you a free piece of advice. Ask Farnham, he was there. |"),
 	    true, TSFX_PEGBOY3 },
-	{ "Abandon your foolish quest. All that awaits you is the wrath of my Master! You are too late to save the child. Now you will join him in Hell! |",
+	{ N_("Abandon your foolish quest. All that awaits you is the wrath of my Master! You are too late to save the child. Now you will join him in Hell! |"),
 	    false, USFX_LAZ1 },
-	{ " |", false, USFX_LAZ1 },
-	{ "Hmm, I don't know what I can really tell you about this that will be of any help. The water that fills our wells comes from an underground spring. I have heard of a tunnel that leads to a great lake - perhaps they are one and the same. Unfortunately, I do not know what would cause our water supply to be tainted. |",
+	{ N_(" |"), false, USFX_LAZ1 },
```
Comment (AJenbo) [inline Source/textdat.cpp:85]:
```suggestion
	{ " |", false, USFX_LAZ1 },
```

File: Source/textdat.cpp [hunk 8]
```diff
@@ -4,495 +4,496 @@
...
 	    true, TSFX_PEGBOY10 },
-	{ "Please, listen to me. The Archbishop Lazarus, he led us down here to find the lost prince. The bastard led us into a trap! Now everyone is dead... killed by a demon he called the Butcher. Avenge us! Find this Butcher and slay him so that our souls may finally rest... |",
+	{ N_("Please, listen to me. The Archbishop Lazarus, he led us down here to find the lost prince. The bastard led us into a trap! Now everyone is dead... killed by a demon he called the Butcher. Avenge us! Find this Butcher and slay him so that our souls may finally rest... |"),
 	    true, TSFX_WOUND },
-	{ " |", true, USFX_CLEAVER },
-	{ "You recite an interesting rhyme written in a style that reminds me of other works. Let me think now - what was it?\n \n...Darkness shrouds the Hidden. Eyes glowing unseen with only the sounds of razor claws briefly scraping to torment those poor souls who have been made sightless for all eternity. The prison for those so damned is named the Halls of the Blind... |",
+	{ N_(" |"), true, USFX_CLEAVER },
```
Comment (AJenbo) [inline Source/textdat.cpp:139]: We can add this once we have support for subtitles and the text has been transcribed.
```suggestion
	{ " |", true, USFX_CLEAVER },
```

File: Source/textdat.cpp [hunk 9]
```diff
@@ -4,495 +4,496 @@
...
+	{ N_("The story of the magic armor called Valor is something I often heard the boys talk about. You had better ask one of the men in the village. |"),
 	    true, TSFX_BMAID13 },
-	{ "The armor known as Valor could be what tips the scales in your favor. I will tell you that many have looked for it - including myself. Arkaine hid it well, my friend, and it will take more than a bit of luck to unlock the secrets that have kept it concealed oh, lo these many years. |",
+	{ N_("The armor known as Valor could be what tips the scales in your favor. I will tell you that many have looked for it - including myself. Arkaine hid it well, my friend, and it will take more than a bit of luck to unlock the secrets that have kept it concealed oh, lo these many years. |"),
 	    true, TSFX_SMITH14 },
-	{ "Zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz... |", true, TSFX_DRUNK15 },
-	{ "Should you find these Stones of Blood, use them carefully. \n \nThe way is fraught with danger and your only hope rests within your self trust. |",
+	{ N_("Zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz... |"), true, TSFX_DRUNK15 },
```
Comment (AJenbo) [inline Source/textdat.cpp:208]: @qndel will go "Chrrrrrrrrr" when he sees this!

File: Source/textdat.cpp [hunk 10]
```diff
@@ -4,495 +4,496 @@
...
+	{ N_("I can see what you see not.\nVision milky then eyes rot.\nWhen you turn they will be gone,\nWhispering their hidden song.\nThen you see what cannot be,\nShadows move where light should be.\nOut of darkness, out of mind,\nCast down into the Halls of the Blind. |\n"),
 	    true, PS_ROGUE11 },
-	{ "The armories of Hell are home to the Warlord of Blood. In his wake lay the mutilated bodies of thousands. Angels and men alike have been cut down to fulfill his endless sacrifices to the Dark ones who scream for one thing - blood. |",
+	{ N_("The armories of Hell are home to the Warlord of Blood. In his wake lay the mutilated bodies of thousands. Angels and men alike have been cut down to fulfill his endless sacrifices to the Dark ones who scream for one thing - blood. |"),
 	    true, PS_ROGUE12 },
-	{ " |", false, TSFX_COW1 },
-	{ " |", false, TSFX_COW2 },
+	{ N_(" |"), false, TSFX_COW1 },
+	{ N_(" |"), false, TSFX_COW2 },
```
Comment (AJenbo) [inline Source/textdat.cpp:496]:
```suggestion
	{ " |", false, TSFX_COW1 },
	{ " |", false, TSFX_COW2 },
```

File: Source/textdat.cpp (hunk 9 above)
Comment (john-tornblom) [reply Source/textdat.cpp:208]: This was actually translated in the Swedish PS1 PAL version (same string value though) :D

File: Source/textdat.cpp [hunk 11]
```diff
@@ -501,158 +502,158 @@ const TextDataStruct alltext[] = {
...
-	{ " |", true, USFX_NAKRUL3 },
-	{ " |", true, USFX_NAKRUL4 },
-	{ " |", true, USFX_NAKRUL5 },
-	{ "And in the year of the Golden Light, it was so decreed that a great Cathedral be raised.  The cornerstone of this holy place was to be carved from the translucent stone Antyrael, named for the Angel who shared his power with the Horadrim.  \n \nIn the Year of Drawing Shadows, the ground shook and the Cathedral shattered and fell.  As the building of catacombs and castles began and man stood against the ravages of the Sin War, the ruins were scavenged for their stones.  And so it was that the cornerstone vanished from the eyes of man. \n \nThe stone was of this world -- and of all worlds -- as the Light is both within all things and beyond all things. Light and unity are the products of this holy foundation, a unity of purpose and a unity of possession.|",
+	{ N_(" |"), true, USFX_DEFILER8 },
+	{ N_(" |"), true, USFX_NAKRUL1 },
+	{ N_(" |"), true, USFX_NAKRUL2 },
+	{ N_(" |"), true, USFX_NAKRUL3 },
+	{ N_(" |"), true, USFX_NAKRUL4 },
+	{ N_(" |"), true, USFX_NAKRUL5 },
```
Comment (AJenbo) [inline Source/textdat.cpp:591]:
```suggestion
	{ " |", true, USFX_DEFILER8 },
	{ " |", true, USFX_NAKRUL1 },
	{ " |", true, USFX_NAKRUL2 },
	{ " |", true, USFX_NAKRUL3 },
	{ " |", true, USFX_NAKRUL4 },
	{ " |", true, USFX_NAKRUL5 },
```

File: Source/textdat.cpp [hunk 12]
```diff
@@ -501,158 +502,158 @@ const TextDataStruct alltext[] = {
...
+	{ N_("Efficio Obitus Ut Inimicus. |"), true, PS_MAGE56 },
+	{ N_("In Spiritu Sanctum. |"), true, PS_ROGUE54 },
+	{ N_("Praedictum Otium. |"), true, PS_ROGUE55 },
+	{ N_("Efficio Obitus Ut Inimicus. |"), true, PS_ROGUE56 },
+	{ N_("In Spiritu Sanctum. |"), true, PS_ROGUE54 },
+	{ N_("Praedictum Otium. |"), true, PS_ROGUE55 },
+	{ N_("Efficio Obitus Ut Inimicus. |"), true, PS_ROGUE56 },
 	/**
 	{ "Please help....! I barely escaped from....... The Butcher...! He killed.... My wife... My children! I beg of you...... In the name of God......... Avenge them......... |", true, TSFX_DEADGUY },
 	{ "*retching* Listen, listen. I don't even like worms. Don't tell me about yer worms! I don't want yer worms! Nope, no thanks! *slight hiccup/gag* No worms for me.", true, TSFX_DRUNK5 },
```
Comment (AJenbo) [inline Source/textdat.cpp:659]:
```suggestion
	{ N_"Please help....! I barely escaped from....... The Butcher...! He killed.... My wife... My children! I beg of you...... In the name of God......... Avenge them......... |"), true, TSFX_DEADGUY },
	{ N_("*retching* Listen, listen. I don't even like worms. Don't tell me about yer worms! I don't want yer worms! Nope, no thanks! *slight hiccup/gag* No worms for me."), true, TSFX_DRUNK5 },

```

File: Source/towners.cpp [hunk 13]
```diff
@@ -453,7 +453,7 @@ void InitGirl()
 	}
 	towners[numtowners]._tNFrames = 20;
 	NewTownerAnim(numtowners, towners[numtowners]._tNAnim[DIR_S], towners[numtowners]._tNFrames, 6);
-	strcpy(towners[numtowners]._tName, "Celia");
+	strcpy(towners[numtowners]._tName, _("Celia"));
```
Comment (AJenbo) [inline Source/towners.cpp:456]: Name
```suggestion
	strcpy(towners[numtowners]._tName, "Celia");
```

File: Source/textdat.cpp (hunk 9 above)
Comment (AJenbo) [reply Source/textdat.cpp:208]: Yeah, not pointing it out as a mistake, just making an internal joke :)

**Review by AJenbo — CHANGES_REQUESTED** [review]
Looks good, but don't mark names and non texts for translation :)

File: Source/textdat.cpp (hunk 9 above)
Comment (john-tornblom) [reply Source/textdat.cpp:208]: :)

File: Source/textdat.cpp [hunk 14]
```diff
@@ -501,161 +502,161 @@ const TextDataStruct alltext[] = {
...
+	{ N_("Efficio Obitus Ut Inimicus. |"), true, PS_ROGUE56 },
+	{ N_("In Spiritu Sanctum. |"), true, PS_ROGUE54 },
+	{ N_("Praedictum Otium. |"), true, PS_ROGUE55 },
+	{ N_("Efficio Obitus Ut Inimicus. |"), true, PS_ROGUE56 },
 	/**
-	{ "Please help....! I barely escaped from....... The Butcher...! He killed.... My wife... My children! I beg of you...... In the name of God......... Avenge them......... |", true, TSFX_DEADGUY },
-	{ "*retching* Listen, listen. I don't even like worms. Don't tell me about yer worms! I don't want yer worms! Nope, no thanks! *slight hiccup/gag* No worms for me.", true, TSFX_DRUNK5 },
+	{ N_"Please help....! I barely escaped from....... The Butcher...! He killed.... My wife... My children! I beg of you...... In the name of God......... Avenge them......... |"), true, TSFX_DEADGUY },
+	{ N_("*retching* Listen, listen. I don't even like worms. Don't tell me about yer worms! I don't want yer worms! Nope, no thanks! *slight hiccup/gag* No worms for me."), true, TSFX_DRUNK5 },
 	{ "*SNORE*", true, TSFX_DRUNK6 },
```
Comment (AJenbo) [inline Source/textdat.cpp:660]: oh look there is more
```suggestion
	{ _("*SNORE*"), true, TSFX_DRUNK6 },
```
and more below this one...

**Review by AJenbo — APPROVED** [review]

**Comment by john-tornblom** [comment]
I was a bit sloppy in my own review, and there might still be a couple of redundant translations. I also ignored strings that seem to go to stdout, and error messages that I seemed unlikely to me.

**Comment by AJenbo** [comment]
Yeah I also think that at least for now we should not translate system errors. It can be really frustrating to get a screen shot of a German error message :D But user errors and the likes should be translated since they are intended for helping the user figure out how to correct there mistake.

File: Source/textdat.cpp (hunk 14 above)
Comment (AJenbo) [reply Source/textdat.cpp]: ... I'm not going to point out ever one of the renaming 125 messages missing from this point.

**Review by AJenbo — APPROVED** [review]
perfect

**Comment by john-tornblom** [comment]
Cool, I think we got most of the strings covered now. I ran the script again on master, and pushed the result to https://github.com/john-tornblom/devilutionX/tree/all-strings-translated

Obviously, this branch won't work, but you can have a look at the strings I missed. 
Take a look at: support_lines.cpp, diablo.cpp, monsterdat.cpp, monster.cpp, and player.cpp
I susspect these files might need more translations, but I cannot tell for sure myself.

**Pull request merged**